README.md

```diff
@@ -1,48 +1,207 @@
+
+You can check one attribute at a time by passing the name to `errors` as a key,
+like so:
+
+```ruby
+person.errors[:name]
+```
+
+
+# Displaying Validation Errors in Views
+
+See [Section 8][ar_validations_8] of the Rails Guide for an example of how to
+use the `ActiveModel::Errors#full_messages()` helper, reproduced here for
+convenience:
+
+[ar_validations_8]: http://guides.rubyonrails.org/active_record_validations.html#displaying-validation-errors-in-views
+
+```erb
+<% if @article.errors.any? %>
+  <div id="error_explanation">
+    <h2>
+      <%= pluralize(@article.errors.count, "error") %>
+      prohibited this article from being saved:
+    </h2>
+
+    <ul>
+    <% @article.errors.full_messages.each do |msg| %>
+      <li><%= msg %></li>
+    <% end %>
+    </ul>
+  </div>
+<% end %>
+```
 
-Introduce the concept of validation, our models protect our DB and can inforce requirements for the attributes.
+This constructs more complete-looking sentences from the more terse messages
+available in `errors.messages`.
 
-The validates macro.
 
-Validates presence :name
+# Custom Validators
 
-When do validations run?
+There are three ways to implement custom validators, with examples in [Section
+6][ar_validators_6] of the Rails Guide:
 
-Looking at a model with a validation error
+- Subclassing `ActiveModel::Validator` and invoking with `#validates_with()`
 
-Trying to save invalid models.
+  This approach is best when you want to do a whole bunch of validations on
+  several different models. You can just call `validates_with MyValidator` on
+  each of them.
 
-Looking at instance.errors
+- Subclassing `ActiveModel::EachValidator` and invoking with an inflected key in
+  the options hash
 
-instance.errors[:field] and instance.errros.on? (if that's still a thing)
+  This is best for validating a single attribute on one model, especially one
+  that you're already using builtin validators for.
 
-full_messages
+  For example, in the Rails Guide, they define `EmailValidator` and then pass
+  the `email: true` key-value pair to `#validates()` to invoke it.
 
-error on a specific field, the error parts (field and message)
+- Calling `#validate()` (that's "validate" *without* an "s") with the name of an
+  instance method
 
-common validations, presence, minimum, maximum, etc, message configurations, etc.
+  Use this approach when you're not sure which to use. If you end up needing to
+  use the same validation logic on a different model, you can easily extract the
+  instance method into one of the ActiveModel classes and use `#validates()` or
+  `#validates_with()` instead.
 
-adding a custom validation via def validate or our own validation macro
+So, to recap:
 
-validates :custom_requirement
+- `Validator` and `validates_with` for doing many validations in one pass.
+- `EachValidator` and `validates` for validating one specific attribute.
+- `validate` for quick custom validations that you can extract later.
```

Comment:
Students love going for the most complex solution first because they feel it's the best, but really we want them to start at the easy ones and build up to the complex. Re-inforce that you should start with a instance method first!